code/game/turfs/open/_open.dm

```diff
@@ -14,53 +14,14 @@
 	var/clawfootstep = null
 	var/heavyfootstep = null
 
-/turf/open/CanPass(atom/movable/A, turf/T)
-	if(isliving(A))
-		var/turf/AT = get_turf(A)
-		if(AT && AT.turf_height - turf_height <= -TURF_HEIGHT_BLOCK_THRESHOLD)
-			return FALSE
-	return ..()
-
 /turf/open/Exit(atom/movable/mover, atom/newloc)
 	. = ..()
-	if(. && isliving(mover) && mover.has_gravity() && isturf(newloc))
+	if(. && isliving(mover) && mover.has_gravity() && isturf(newloc) && can_zFall())
```

Comment:
can_zFall checks if the turf is able to zPassOut, you want to check can_zFall on the mob, however that zFall also checks the turf below, so you'd have to make another proc that splits the logic.

also make sure you get `/mob/living/carbon/can_zFall`, `/mob/living/can_zFall` and `/atom/movable/can_zFall`.

might be best just to ignore for now? idk

Reply:
Swapped to checking for mob, the rest seems like unrelated rambling.